File: C-Abi-Bridge-Aot/Tests/CAbiBridge.Aot.Cpp.Tests/crypto_aes_gcm_test.cpp
#include "pch.h"

#include <iostream>
#include "crypto_aes_gcm_test.h"


namespace michele::natale::Tests
{
  static void start_native_aes_gcm_crypto(int rounds)
  {
    test_aes_gcm_file(rounds);
    test_aes_gcm_bytes(rounds);
    test_aes_gcm_bytes_stress();

    std::cout << "\n";
  }


  void start_aes_gcm_native(int rounds)
  {
    start_native_aes_gcm_crypto(rounds);
  }
}

File: C-Abi-Bridge-Aot/Tests/CAbiBridgeAot.Tests/TestCrypto/TestCryptoCryption/TestCryptoSymmetric/TestCryptoAesGcm/CryptoAesGcmTest.cs


namespace michele.natale.Tests;

internal sealed partial class CryptoAesGcmTest
{
  public static void StartNative(int rounds)
  {
    TestAesGcmFile(rounds);

    TestAesGcmBytes(rounds);

    TestAesGcmBytesStress();

    Console.WriteLine();
  }

  public static void StartManaged(int rounds)
  {

    //Nur für Testzwecken der Managed-Methoden,
    //da mit 'UnmanagedCallersOnly' der Zugriff
    //zu Testszwecken nicht gewährt wird.
    //Bitte für die Nativen Tests dekommentieren!!

    //Only for testing purposes of managed-methods,
    //as access for testing purposes is not granted
    //with ‘UnmanagedCallersOnly’.
    //Please uncomment for native tests!

    TestAesGcmFileManaged(rounds);

    TestAesGcmBytesManaged(rounds);

    TestAesGcmBytesStressManaged();

    Console.WriteLine();
  }
}
